packages/css/src/plugins/expand-shorthands/text-decoration.tsx

```diff
@@ -0,0 +1,55 @@
+import { Node } from 'postcss-values-parser';
+import { ConversionFunction } from './types';
+import { globalValues, isColor } from './utils';
+
+/**
+ * https://developer.mozilla.org/en-US/docs/Web/CSS/text-decoration
+ */
+export const textDecoration: ConversionFunction = (value) => {
+  const [left, middle, right] = value.nodes;
+  const lineValues = [...globalValues, 'none', 'underline', 'overline', 'line-through', 'blink'];
+  const styleValues = [...globalValues, 'solid', 'double', 'dotted', 'dashed', 'wavy'];
+  const lineValue: string[] = [];
+  let colorValue = '';
+  let styleValue = '';
+
+  /**
+   * Extracts values from a node and mutates variables in scope.
+   * If it returns `true` we should bail out and return no nodes.
+   *
+   * @param node
+   */
+  const extractValues = (node: Node): boolean => {
+    if (node && node.type === 'word') {
+      if (lineValues.includes(node.value)) {
+        if (lineValue.length === 0) {
+          lineValue.push(node.value);
+        } else if (!lineValue.includes(node.value)) {
+          lineValue.push(node.value);
+        } else {
```

Comment:
hmmm..looks hard though..hehehe..Following looks repeated `lineValue.push(node.value);`. But its ok to skip this one.

```suggestion
        if (lineValue.length === 0 || !lineValue.includes(node.value)) {
          lineValue.push(node.value);
        } else {
```

And its ok if refactoring is not possible. Lets merge this one right now. We can anyhow comeback to this piece of code later. Because code evolves as time goes one. And "I don't like premature abstraction" because "code duplication is far cheaper than the wrong abstraction" 😌

Reply:
done